network/discovery/src/rest.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::validator_set::extract_validator_set_updates;
+use crate::DiscoveryError;
+use aptos_config::config::PeerSet;
+use aptos_config::network_id::NetworkContext;
+use aptos_logger::info;
+use aptos_time_service::{Interval, TimeService, TimeServiceTrait};
+use aptos_types::account_address::AccountAddress;
+use aptos_types::on_chain_config::ValidatorSet;
+use futures::executor::block_on;
+use futures::Stream;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+/// A discovery stream that uses the REST client to determine the validator
+/// set nodes.  Useful for when genesis is significantly far behind in time
+pub struct RestStream {
+    network_context: NetworkContext,
+    rest_client: aptos_rest_client::Client,
+    interval: Pin<Box<Interval>>,
+}
```

Comment:
I don't know where we are with our manager mode for the blockchain, but if we are up to date, we're just making a wasted query to our rest service. it might be good to specify that the timer needs to be on the order of 60 or even 600 seconds so we don't end up with 4000 or 20000 tps if this is enabled by default.

Reply:
if we are connected, we could even consider disabling the poll and only kicking this off again if we are live

Reply:
Yeah, I think this is fair.  Probably should do an "initialize" time, and then stop doing it repeatedly.

This is really only for the "startup" period

Reply:
Now this backs off 10x the time used for interval time, once it's loaded once.